### PR TITLE
Add ability to specify signature suite.

### PR DESCRIPTION
**Pull request by msporny**

### CODE REVIEW

File: lib/ZcapClient.js [hunk 1]
```diff
@@ -92,13 +102,16 @@ export class ZcapClient {
...
+   *   suite to use when digitally signing the delegation. Default: The
+   *   signature suite that was used when initializing the ZcapClient.
    *
    * @returns {Promise<object>} - A promise that resolves to a delegated
    *   capability.
    */
   async delegate({
     url, capability, targetDelegate, invocationTarget, expires,
-    allowedActions = []
+    allowedActions = [], suite
```
Comment (dlongley) [inline lib/ZcapClient.js:114]: Perhaps this should be an *instance* of a suite? Is it safe to assume that all suite instances can be constructed with only a `signer` and `verificationMethod`? Alternatively, we could accept either a `suiteClass` or `suite` here.

File: lib/ZcapClient.js [hunk 2]
```diff
@@ -46,16 +53,19 @@ export class ZcapClient {
...
    *   `didDocument` and `keyPairs` must be provided.
+   * @param {object} [options.suite] - The default signature suite to use when
+   *   digitally signing delegations. Default: Ed25519Signature2018.
    *
    * @returns {ZcapClient} - The new ZcapClient instance.
    */
   constructor({
     baseUrl, defaultHeaders = {}, agent, didDocument, keyPairs,
-    invocationSigner, delegationSigner
+    invocationSigner, delegationSigner, suite = Ed25519Signature2018
```
Comment (dlongley) [inline lib/ZcapClient.js:63]: Shouldn't this be something like `suiteClass`? This isn't an instance of a suite.

File: lib/ZcapClient.js [hunk 3]
```diff
@@ -46,16 +53,19 @@ export class ZcapClient {
    *   `.sign()` function and `id` and `controller` properties that will be
    *   used for signing requests. `invocationSigner` and `delegationSigner`, or
    *   `didDocument` and `keyPairs` must be provided.
+   * @param {object} [options.suite] - The default signature suite to use when
```
Comment (dlongley) [inline lib/ZcapClient.js:56]: Whatever is decided here (suite or suiteClass), `object` isn't correct for the docs.

**Comment by dmitrizagidulin** [comment]
I agree with @dlongley's comments. @msporny - would you like me to take a stab at addressing those?

**Comment by dlongley** [comment]
Closing this in favor of some other work that will also resolve document loader issues.

**Pull request closed without merging**